### PR TITLE
Python: Start modelling the standard library using MaD

### PR DESCRIPTION
**Pull request by yoff**
I am thinking to add a collective change note at the end mentioning "several standard library functions" or some such. Let me know if you want incremental change notes instead..

### CODE REVIEW

**Comment by RasmusWL** [comment]
> I am thinking to add a collective change note at the end mentioning "several standard library functions" or some such. Let me know if you want incremental change notes instead..

sounds fine :+1:

File: python/ql/lib/ext/StdLib.model.yml [hunk 1]
```diff
@@ -12,8 +12,26 @@ extensions:
...
-    data: []
-
+    data:
+      # See https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote
+      - ["urllib", "Member[parse].Member[quote]", "Argument[0,string:]", "ReturnValue", "taint"]
+      # See https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote_plus
+      - ["urllib", "Member[parse].Member[quote_plus]", "Argument[0,string:]", "ReturnValue", "taint"]
+      # See https://epydoc.sourceforge.net/stdlib/urllib-module.html
+      - ["urllib", "Member[parse].Member[splitquery]", "Argument[0,url:]", "ReturnValue.TupleElement[0]", "taint"]
+      - ["urllib", "Member[parse].Member[splitquery]", "Argument[0,url:]", "ReturnValue.TupleElement[1]", "taint"]
```
Comment (RasmusWL) [inline python/ql/lib/ext/StdLib.model.yml:22]: Would be nice if you could do `"ReturnValue.TupleElement[0,1]"`

**Review by RasmusWL — COMMENTED** [review]
Can you highlight why you put the models where you did? that is, why not put them under `python/ql/lib/semmle/python/frameworks/`?

Also, did you consider whether `quote`/`qoute_plus` should be sanitizers for any queries? (I can't think of any myself :thinking:)

otherwise LGTM

File: python/ql/lib/ext/StdLib.model.yml (hunk 1 above)
Comment (yoff) [reply python/ql/lib/ext/StdLib.model.yml:22]: Hm, maybe we already can. I should add a test... :-)

**Comment by yoff** [comment]
> Can you highlight why you put the models where you did? that is, why not put them under `python/ql/lib/semmle/python/frameworks/`?

I followed C# (which has the same structure as Java and Go) but that was just because I did not realize that we already have a natural place for it...I will move it :-)
 
> Also, did you consider whether `quote`/`qoute_plus` should be sanitizers for any queries? (I can't think of any myself 🤔)

Yes, but I did not immediately see any. We have `shlex.quote` as a sanitizer for `UsafeShellCommandConstruction` but that is quite specialised to shells. It did make me wonder what takes precedence, a MaD model or a sanitizer. I believe a sanitizer will, but I will check..

**Comment by RasmusWL** [comment]
Sorry for the ping, that's my fault :grimacing:

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**